js/components/SocialShare/index.js

```diff
@@ -0,0 +1,58 @@
+/* global require */
+import React, { Component, PropTypes } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+} from 'react-native';
+import Share from 'react-native-share';
+
+export default class SocialShare extends Component {
+  constructor(props) {
+    super(props);
+
+    this.share = this.share.bind(this);
+  }
+
+  share() {
+    let shareOptions = this.props.itemId ? 
+      this.fromStore() : 
+      Object.assign({}, this.props);
+
+    Share.open(shareOptions);
+  }
+
+  fromStore() {
+    // TODO: Retrieve the share options from an item in the store
+    return {
+      message: 'Shared from Magnet',
+    };
+  }
+
+  render() {
+    return(
+      <TouchableOpacity
+        style = { styles.container }
+        onPress = { this.share }>
+        <Image source = { require('./assets/share.png') }/>
+      </TouchableOpacity>
+    );
+  }
+}
+
+SocialShare.propTypes = {
+  url: PropTypes.string,
+  type: PropTypes.string,
+  message: PropTypes.string,
+  title: PropTypes.string,
+  subject: PropTypes.string,
+  itemId: PropTypes.number,
+};
+
+const styles = StyleSheet.create({
+  container: {
+    height: 24,
+    width: 24,
+    margin: 10,
```

Comment:
Margin shouldn't be defined in the component itself but defined in the context it's used.

```js
<SocialShare style={ margin: 10 }/>
```

To do this you'll need to use the `this.props.style` in the `render()`

```js
render() {
  return(
    <TouchableOpacity
      style = {[ styles.container, this.props.style ]}
      ...
    </TouchableOpacity>
  );
}
```

And add it to the `SocialShare.propTypes`:

```js
SocialShare.propTypes = {
  View.propTypes.style,
  ...
};
```